reference/ClosureBlog.md

```diff
@@ -0,0 +1,65 @@
+
+While FaaS systems have become heavily adopted by many, the process of getting up and running with them can still feel overly complex compared to the ease at which they work at runtime.  Looking at AWS' FaaS offering [Lambda](https://aws.amazon.com/lambda/), there is a distinct split between doing all the work to configure the Lambda runtime itself (i.e. how much memory to use, what environment variables should be present, etc.), versus how to write and maintain the code that will execute *in* the Lambda itself when triggered.  For AWS, the code itself needs to written and packaged up with all its dependencies into a single zip-file, stored as a blob that is then placed in [AWS S3](https://aws.amazon.com/s3/).  This blob is then referenced by Lambda so that it can be loaded on-demand when necessary.  Updating just a single character of this code then requires fully repackaging and reuploading everything.  Compounding all of that, this code is necessary independent of any other pieces of ones cloud-infrastructure.  If the code for the Lambda needs to read or write to an S3 bucket, it needs a good way to have that bucket's information passed to it.  Same with publishing to an [AWS SNS](https://aws.amazon.com/sns/) topic, or any other relevant resources needed.  These same issues also apply if creating [FunctionApps](https://docs.microsoft.com/en-us/azure/azure-functions/functions-overview) using Nodejs on Azure.
+
+At Pulumi, we thought there could be a better way to do this.  We already wanted to supply an easy way for developers to simply spin up their cloud infrastructure using languages and tooling they were already familiar with.  However, when it came to FaaS offerings, things still felt a bit too onerous at first.  While you could certainly create an AWS Lambda with code like `new aws.lambda.Function("myfunc")`, you would still have to do all the work external to your Pulumi application to define your code, its dependencies, ways to pass necessary information into it, as well as potentially needing to have build tasks running to keep this all up to date.  That's a lot of extra, out-of-band, effort.  Doubly-so since a Pulumi application is *already* code!  Why have to jump out of that code just to create another bit of code that you need for running your Cloud application?  From a desire to do better here, the "closure serialized function" was borne.  However, "closure serialized function" is a pretty geeky mouthful, so I like just calling them "magic functions".  Before diving more deeply into them, let's take a look at how they look in action (note: some of this example depends on  API changes that will be present in an upcoming release):
```

Comment:
Maybe trim this down?

At Pulumi, we thought there could be a better way to do this. From a desire to do better here, the "closure serialized function" was borne.  However, "closure serialized function" is a pretty geeky mouthful, so I like just calling them "magic functions".  Before diving more deeply into them, let's take a look at how they look in action (note: some of this example depends on  API changes that will be present in an upcoming release):